src/yetibot/commands/karma/specs.clj

```diff
@@ -0,0 +1,34 @@
+(ns yetibot.commands.karma.specs
+  (:require [clojure.spec.alpha :as s]))
+
+;; Common specs we could relocatexo
+(s/def ::name string?)
+(s/def ::id string?)
+(s/def ::user (s/keys :req-un [::name ::id]))
+
+(s/def ::adapter #{:irc :slack})
+(s/def ::chat-source (s/keys :req-un [::adapter]))
+
+(s/def ::ctx (s/keys :req-un [::user ::chat-source]))
+
+;; Karma-specific specs
+(s/def ::user-id (s/and string? #(re-matches #"@?\w[-\w]*\w" %)))
+(s/def ::action (s/and string? (s/or :positive #(= "++" %)
+                                     :negative #(= "--" %))))
+(s/def ::note string?)
+
+;; run-time fn data validators
+
+;; get-score
+(s/def :yetibot.commands.karma.specs.get-score/match string?)
+(s/def ::get-score-ctx
+  (s/merge ::ctx (s/keys :req-un [:yetibot.commands.karma.specs.get-score/match])))
+
```

Comment:
@devth let's discuss my thinking behind these long ns-qualified keywords.  They're ugly; would be nice to find a better way.

Reply:
Woke up thinking about this alternate approach.  It's interesting but might be overly clever and opaque?  I'm not sure - manipulating namespaces is mostly new to me.  Nonetheless, this was fun and cemented some things for me.

I'll write up a docstring with details a little later today.

Reply:
Very interesting idea 😁 

I think it's a bit of a departure from convention. We shouldn't have to reach for tools this powerful/meta for a pretty ordinary spec use case. 

I'd probably instead name specs like:

```clojure
(s/def ::get-score.match string?)
```